### PR TITLE
Kedro Viz Lite User Warning UI

### LINKED ISSUE #2058: Kedro Viz Lite User Warnings
## Description

This ticket is to address next steps in https://github.com/kedro-org/kedro-viz/pull/1966

1. `FE notice with a warning of missing imports` suggestion from TD on Lightweight Kedro Viz implementation
2. `CLI showing a warning message with missing imports` (already [implemented](https://github.com/kedro-org/kedro-viz/pull/1966#issuecomment-2307429604) but needs confirmation from design on the wording. @stephkaiser )
3. `MemoryDataset to custom ImportErrorDataset in-case of missing dataset dependencies in the metadata panel`. Since the focus is mostly on flowchart, I am not sure if we should create a custom dataset or leave this as MemoryDataset. (need to discuss this during backlog grooming)

## Context

We mocked missing dependencies in Kedro project when users run `kedro viz --lite`. This ticket aims at letting the users aware of the mocked dependencies both in UI and CLI

## Possible Implementation

1. In CLI, have a logger warning message to inform users of the missing dependencies. Example:

```
[08/23/24 16:33:59] WARNING  Kedro-Viz has mocked the following             data_loader.py:173
                             dependencies for lite-mode.                                      
                             ['sklearn.base', 'matplotlib.pyplot',                            
                             'matplotlib', 'seaborn', 'PIL',                                  
                             'sklearn.model_selection', 'sklearn',                            
                             'sklearn.metrics']                                               
                             In order to get a complete experience of Viz,                    
                             please install the missing Kedro project                         
                             dependencies 
```

2. In UI, have a hidden banner or something similar to [this](https://docs.servicenow.com/bundle/utah-platform-user-interface/page/administer/navigation-and-ui/concept/next-experience-banner-announcements.html) website and Kedro theme

3. Leave the MemoryDataset as-is or implement a custom dataset and discuss on what all functionalities to include in the dataset (mostly leaving the bare-minimum) to show up in metadata panel

## Checklist

- [x] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #2058 

## Development notes

**Backend changes:**
* Refactored PackageCompatibilityAPI response to MetadataAPIResponse
* Updated the API endpoint from [GET] "/package-compatibilities" to "/metadata" and updated the response model
* Moved`get_package_compatibilities` to utils
* Created models to hold `PackageCompatibility` and `Metadata` for the application

**Frontend changes:**
* Updated function calls to fetchMetadata as per the backend updated endpoint
* Added `showBanner` to the redux store and created generic actions, reducers to extend banner functionality in future
* Created Banner components and added styles
* Added tests for banner component

Thank you @jitu5 for helping out in resolving the redux states 💯 

image<br>

image

## QA notes

* All tests should pass
* You should see the banner when there are missing dependencies in your python env and you do `kedro viz --lite`. The banner should not appear again for the rest of the browser session once you close it.
* The `Learn More` button click should take you to the docs for `--lite`
* There should not be any changes with the `package_compatibility` feature we have in deploy and ET.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by stephkaiser** [comment]
thanks Ravi, that was super quick! 

some minor design QA comments from me:
- font size for the title and description should be 16, looks like its currently 14
- height of the banner should be 68px max, looks like its 77px currently

Screenshot 2024-09-13 at 17 50 14

thank you!

File: src/config.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ export const localStorageFlowchartLink = 'KedroViz-link-to-flowchart';
 export const localStorageMetricsSelect = 'KedroViz-metrics-chart-select';
 export const localStorageRunsMetadata = 'KedroViz-runs-metadata';
 export const localStorageShareableUrl = 'KedroViz-shareable-url';
+export const localStorageBannerStatus = 'KedroViz-banners';
```
[the remark this answers is not in the document]
Comment (rashidakanchwala) [reply src/config.js:8]: i like that. maybe we can refactor localStorage in a future ticket.

**Comment by Huongg** [comment]
this is pretty cool, thanks @ravi-kumar-pilla. Just a quick question about the setup, I'm trying to run it locally, but i have the error `Error: No such option: --lite Did you mean --pipeline?` when i ran `kedro viz --lite`

I've tested with your branch with a new environment. Is there anything else I need to do?

Screenshot 2024-09-17 at 11 15 31

**Review by rashidakanchwala — APPROVED** [review]
It works well. and I have looked through the code quickly. There are some possible refactorings we can focus on FE (not caused by this PR but previous ones as well) in the future which we can look into after this release. Thanks a lott @ravi-kumar-pilla , passing data through `api/metadata` looks great

**Review by stephkaiser — APPROVED** [review]

File: src/config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply src/config.js:8]: yes we can have a common key in future tickets

File: src/styles/_variables.scss [hunk 2]
```diff
@@ -31,6 +31,7 @@ $run-list-controls-height: 95px;
 $z-index-metadata-panel: 3;
 $z-index-metadata-code: 3;
 $z-index-MuiTreeItem-icon: 1;
+$z-index-banner: 6;
```
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply src/styles/_variables.scss:34]: Yes there is global toolbar and feature hints with z-index:5 . Another reason I thought is we should always have our banners on top of other components, so made it a 6.

**Review by Huongg — APPROVED** [review]
its working now for me, thanks @ravi-kumar-pilla
